provisioning/service/src/EnrollmentGroupsClient.cs

```diff
@@ -0,0 +1,281 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Threading;
+using Newtonsoft.Json;
+using System.Globalization;
+using System.Net;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Devices.Provisioning.Service
+{
+    /// <summary>
+    /// Subclient of <see cref="ProvisioningServiceClient"/> that handles all enrollment group operations including
+    /// getting/creating/setting/deleting enrollment groups, querying enrollment groups, and getting attestation mechanisms
+    /// for particular enrollment groups.
+    /// </summary>
+    public class EnrollmentGroupsClient
+    {
+        private const string ServiceName = "enrollmentGroups";
+        private const string EnrollmentIdUriFormat = "{0}/{1}?{2}";
+        private const string EnrollmentAttestationName = "attestationmechanism";
+        private const string EnrollmentAttestationUriFormat = "{0}/{1}/{2}?{3}";
+
+        private readonly IContractApiHttp _contractApiHttp;
+        private readonly ServiceConnectionString _serviceConnectionString;
+
+        /// <summary>
+        /// Creates an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        protected EnrollmentGroupsClient()
+        {
+        }
+
+        internal EnrollmentGroupsClient(ServiceConnectionString serviceConnectionString, IContractApiHttp contractApiHttp)
+        {
+            _serviceConnectionString = serviceConnectionString;
+            _contractApiHttp = contractApiHttp;
+        }
+
+        /// <summary>
+        /// Create or update an enrollment group record.
+        /// </summary>
+        /// <remarks>
+        /// This API creates a new enrollment group or update a existed one. All enrollment group in the Device
+        /// Provisioning Service contains a unique identifier called enrollmentGroupId. If this API is called
+        /// with an enrollmentGroupId that already exists, it will replace the existed enrollment group information
+        /// by the new one. On the other hand, if the enrollmentGroupId does not exit, it will be created.
+        ///
+        /// To use the Device Provisioning Service API, you must include the follow package on your application.
```

Comment:
?

Reply:
This was part of the old XML docs I copied over. I'll clean these up a bit, sure.